influxql/parser.go

```diff
@@ -265,6 +267,58 @@ func (p *Parser) parseSetPasswordUserStatement() (*SetPasswordUserStatement, err
 	return stmt, nil
 }
 
+// parseBackfillStatement parses the backfill statement. It assumes that the
+// BACKFILL keyword has already been consumed.
+func (p *Parser) parseBackfillStatement() (*BackfillStatement, error) {
+	stmt := &BackfillStatement{}
+	// Parse the CQ name
+	ident, err := p.parseIdent()
+	if err != nil {
+		return nil, err
+	}
+	stmt.QueryName = ident
+
+	// get the ON keywork
+	tok, pos, lit := p.scanIgnoreWhitespace()
+	if tok != ON {
+		return nil, newParseError(tokstr(tok, lit), []string{"ON"}, pos)
+	}
+
+	// Get the database name
+	ident, err = p.parseIdent()
+	if err != nil {
+		return nil, err
+	}
+	stmt.Database = ident
+
+	// get the FROM keyword
+	tok, pos, lit = p.scanIgnoreWhitespace()
+	if tok != FROM {
+		return nil, newParseError(tokstr(tok, lit), []string{"FROM"}, pos)
+	}
+
+	expr, err := p.ParseExpr()
+	if err != nil {
+		return nil, err
+	}
+	stmt.From = expr
+
+	// Get the UNTIL keyword. This is optional, so if we see EOF, just ignore it.
+	tok, pos, lit = p.scanIgnoreWhitespace()
+	if tok != UNTIL {
```

Comment:
Multiple statements could be separated by a semicolon.

On Thursday, October 22, 2015, otoolep <notifications@github.com> wrote:

> In influxql/parser.go
> <https://github.com/influxdb/influxdb/pull/4528#discussion_r42814999>:
>
> > +
> > +	// get the FROM keyword
> > +	tok, pos, lit = p.scanIgnoreWhitespace()
> > +	if tok != FROM {
> > +		return nil, newParseError(tokstr(tok, lit), []string{"FROM"}, pos)
> > +	}
> > +
> > +	expr, err := p.ParseExpr()
> > +	if err != nil {
> > +		return nil, err
> > +	}
> > +	stmt.From = expr
> > +
> > +	// Get the UNTIL keyword. This is optional, so if we see EOF, just ignore it.
> > +	tok, pos, lit = p.scanIgnoreWhitespace()
> > +	if tok != UNTIL {
>
> If the token is something else other than EOF, isn't that an error?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdb/influxdb/pull/4528/files#r42814999>.
>